File: python/beer-song/beer_song.py
def recite(start, take=1):
  res = []
  for i in range(start, start - take, -1):
    res += ((verse(i)) + [''])
  return res[:-1]

def verse(num):
  res = [f'{num_bottles(num)} of beer on the wall, {num_bottles(num).lower()} of beer.']
  # res1 = ''
  if num == 0:
    res += ['Go to the store and buy some more, ']
    new_bottles = 99
  else:
    pronoun = 'it' if num == 1 else 'one'
    res += [f'Take {pronoun} down and pass it around, ']
    new_bottles = num - 1
  res[1] += f'{num_bottles(new_bottles).lower()} of beer on the wall.'
  return res

def num_bottles(num):
  return ('No more' if num == 0 else str(num)) + (' bottle' if num == 1 else ' bottles')

File: kotlin/beer-song/src/main/kotlin/BeerSong.kt
package BeerSong

fun verses(start: Int, end: Int): String {
  var res = arrayListOf<String>();
  for (i in start downTo end) {
    res.add(verse(i))
  }
  return res.joinToString("\n")
}

fun verse(n: Int): String {
  var numBottles1 = if (n == 0) "No more" else n.toString() 
  var numBottles2 = if (n == 0) "no more" else n.toString()
  var bottles = if (n == 1) "bottle" else "bottles"
  var take = if (n == 1) "it" else "one"
  var remaining = if (n == 1) "no more" else (n - 1).toString()
  var bottles2 = if (n == 2) "bottle" else "bottles"

  var part1 = "$numBottles1 $bottles of beer on the wall, $numBottles2 $bottles of beer.\n"
  var part2 = if (n == 0) "Go to the store and buy some more, 99 bottles of beer on the wall.\n" else 
    "Take $take down and pass it around, $remaining $bottles2 of beer on the wall.\n"
  return part1 + part2
}

